core/corehttp/gateway_handler.go

```diff
@@ -133,6 +133,7 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 
 	w.Header().Set("X-IPFS-Path", urlPath)
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
- `Access-Control-Allow-Credentials` is just for cookies & HTTP-Basic auth, not a more modern / explicit method of authenticating like oAuth.
- why would we likely have authentication on a gateway? wouldn't encryption of the resources themselves be the preferred method of keeping data safe on a p2p network?
- gateway operators wanting to restrict the origins that they allow seems kinda silly, but they can still totally do that with a reverse-proxy like nginx in front of ipfs... I'd be happy to write up a guide if this is something that people actually want to do.
- how would this apply to non-browsers? when you use cURL it doesn't hide the response from you when `null` isn't a whitelisted origin... CORS is enforced by the browser and only in the browser.